main.py

```
@@ -25,6 +25,49 @@ async def build_esp_related_project(project_path: str) -> (str, str):
+        project_path (str): Path where the new ESP-IDF project will be created. 
+                            Must be located directly under the current working directory.
+        project_name (str): Name of the ESP-IDF project to create.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the standard output and standard error messages.
+    """
+    os.makedirs(project_path, exist_ok=True)
+    os.chdir(project_path)
+    export_script = get_export_script()
+    returncode, stdout, stderr = await run_command_async(f"bash -c 'source {export_script} && idf.py create-project --path {project_path} {project_name}'")
+    open('mcp-project-root-path.log', 'w+').write(str((stdout, stderr)))
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)